pandas/tests/reshape/test_tile.py

```diff
@@ -512,6 +512,23 @@ def f():
         tm.assert_numpy_array_equal(
             mask, np.array([False, True, True, True, True]))
 
+    def test_cut_read_only(self):
+        # issue 18773
+        def _test_memview_cut(array_1_writeable, array_2_writeable):
+            array_1 = np.arange(0, 100, 10)
+            array_1.flags.writeable = array_1_writeable
+
+            array_2 = np.arange(0, 100, 10)
+            array_2.flags.writeable = array_2_writeable
+
+            hundred_elements = np.arange(100)
+
+            tm.assert_categorical_equal(cut(hundred_elements, array_1),
+                                        cut(hundred_elements, array_2))
+
+        for (w_1, w_2) in [[True, True], [True, False], [False, False]]:
```

Comment:
can you parametrize this test instead of writing a sub test function

Reply:
sorry yea just pushed a commit for that. should be parametrized now :smile: